qos/evm/check.go

```diff
@@ -50,19 +60,53 @@ func getEndpointCheck(
 }
 
 // withChainIDCheck updates the request context to make an EVM JSON-RPC eth_chainId request.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_chainId"}'
 func withChainIDCheck(requestCtx *requestContext) {
 	requestCtx.jsonrpcReq = buildJSONRPCReq(idChainIDCheck, methodChainID)
 }
 
 // withBlockHeightCheck updates the request context to make an EVM JSON-RPC eth_blockNumber request.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_blockNumber"}'
 func withBlockHeightCheck(requestCtx *requestContext) {
 	requestCtx.jsonrpcReq = buildJSONRPCReq(idBlockNumberCheck, methodBlockNumber)
 }
 
-func buildJSONRPCReq(id int, method jsonrpc.Method) jsonrpc.Request {
-	return jsonrpc.Request{
+// withArchivalBlockCheck updates the request context to make an EVM JSON-RPC eth_getBalance request with a random archival block number.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_getBalance","params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]}'
+func withArchivalBlockCheck(requestCtx *requestContext) {
+	// Get the archival block number from the endpoint store.
+	archivalCheckConfig := requestCtx.endpointStore.serviceState.archivalCheckConfig
+	// Get the current state of the archival check.
+	serviceArchivalState := requestCtx.endpointStore.serviceState.archivalState
+
+	requestCtx.jsonrpcReq = buildJSONRPCReq(
+		idArchivalBlockCheck,
+		methodGetBalance,
+		// Pass params in this order, eg. "params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]
+		// Reference: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getbalance
+		archivalCheckConfig.ContractAddress,
+		serviceArchivalState.blockNumberHex,
```

Comment:
Why aren't we using `archivalCheckConfig.perceivedBlockNumber`  or `archivalCheckConfig.evmArchivalState.blockNumberHex` here?

Need to #PUC so its immediately evident to anyone, even if they only review this small part of the codebase.

Reply:
Also removed as this file has been refactored according to @adshmh 's suggestion here: https://github.com/buildwithgrove/path/pull/194#discussion_r2026804634

IMO it was a great suggestion and improvement.